src/redux/Missions.js

```diff
@@ -0,0 +1,49 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const missionsApi = 'https://api.spacexdata.com/v3/missions';
+
+// fetch missions thunk
+export const fetchMissions = createAsyncThunk(
+  'missions/fetchMissions',
+  async ({ limit }) => {
+    const response = await axios.get(`${missionsApi}?limit=${limit}`);
+    return response.data;
+  },
+);
+const initialState = [];
+
+// mission slice
+const missionsSlice = createSlice({
+  name: 'missions',
+  initialState,
+  reducers: {
+    joinMission: {
+      reducer: (state, action) => state.map((mission) => (mission.mission_id !== action.payload
+        ? mission
+        : { ...mission, reserved: true })),
+    },
+    leaveMission: {
+      reducer: (state, action) => state.map((mission) => (mission.mission_id !== action.payload
+        ? mission
+        : { ...mission, reserved: false })),
+    },
+  },
+
+  extraReducers: {
+    [fetchMissions.fulfilled]: (state, { payload }) => {
```

Comment:
- [ ] Please remove this `state` parameter as **it is not used throughout the function** This helps to prevent some bugs that may arise in the future 😉